### PR TITLE
[ADD]Added web_security_dialog module.

### PR DESCRIPTION
**Pull request by vishal-onnet**

### CODE REVIEW

File: web_security_dialog/LICENSE/LICENSE [hunk 1]
```diff
@@ -0,0 +1,17 @@
+    OpenERP, Open Source Management Solution
+    Copyright (C) 2004-2010 OpenERP SA (<http://www.openerp.com>)
+    Copyright (C) 2011-2017 Serpent Consulting Services Pvt. Ltd. (<http://www.serpentcs.com>).
+
```
Comment (JayVora-SerpentCS) [inline web_security_dialog/LICENSE/LICENSE]: Dear Vishal, please do the push logically.
remove openerp Sa line. Was this module copied by you or made by you?

Comment (JayVora-SerpentCS) [reply web_security_dialog/LICENSE/LICENSE]: Copyright (C) 2017-TODAY Serpent Consulting Services Pvt. Ltd. (<http://www.serpentcs.com>)

Comment (vishal-onnet) [reply web_security_dialog/LICENSE/LICENSE]: Yes sir, I will make the changes right away & will be careful about such things in the future.

File: web_security_dialog/models/res_users.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# See LICENSE file for full copyright and licensing details.
+from odoo import fields, models, api
+
+
+class SecurityDialig(models.Model):
+
+    _inherit = 'res.users'
+
+    security_dialog = fields.Char('Security Code')
```
Comment (ghost) [inline web_security_dialog/models/res_users.py]: Security Field should be dependent on company not on the user.

File: web_security_dialog/models/res_users.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        if(vals.get('userId')):
+            user_object = self.browse(vals.get('userId'))
+            result = user_object.search_read([
+                        ('id', '=', vals.get('userId'))], [current_fields])
+            if(result[0].get(current_fields or False) == vals.get('password')
+               .encode('ascii', 'ignore')):
+                return True
+            else:
+                return False
+        return False
```
Comment (ghost) [inline web_security_dialog/models/res_users.py]: Please optimize code here.

File: web_security_dialog/models/res_users.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# See LICENSE file for full copyright and licensing details.
+from odoo import fields, models, api
+
+
+class SecurityDialig(models.Model):
+
```
Comment (JayVora-SerpentCS) [inline web_security_dialog/models/res_users.py]: Classname --> SecurityDialog

File: web_security_dialog/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    'data': [
+        'views/res_users_security.xml',
+        'views/templates.xml'
+    ],
+    'qweb': ['static/src/xml/web_security_dialog.xml'],
+    'auto_install': False,
+    'installable': True,
+    'application': True,
+}
+
```
Comment (JayVora-SerpentCS) [inline web_security_dialog/__manifest__.py]: The screen web_security_dialog/static/description/img/add_code.png should better have SerpentCS logo

Comment (JayVora-SerpentCS) [reply web_security_dialog/__manifest__.py]: Make sure all images have SerpentCS logo as company logo.

File: web_security_dialog/static/src/js/web_security_dialog.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                                           framework.blockUI();
+                                           var callback = self.validate_user(self.is_dialog_security,pass_value);
+                                           callback.done(function(result) {
+                                               framework.unblockUI();
+                                               if (result) {
+                                                   curr_obj.close();
+                                                   self.click_operation();
+                                               }else {
+                                                   Dialog.alert(self, _t("Password is Wrong"));
+                                                   return;
```
Comment (JayVora-SerpentCS) [inline web_security_dialog/static/src/js/web_security_dialog.js]: the text should be 'Invalid or Wrong Password!'

File: web_security_dialog/static/src/js/web_security_dialog.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+                                               }
+                                           }).fail(function(error) {
+                                               framework.unblockUI();
+                                               Dialog.alert(self, _t("Connection lost"));
```
Comment (JayVora-SerpentCS) [inline web_security_dialog/static/src/js/web_security_dialog.js]: The message should be 'Either the password is wrong or the connection is lost! Contact your Administrator.'

File: web_security_dialog/models/res_users.py (hunk 4 above)
Comment (vishal-onnet) [reply web_security_dialog/models/res_users.py]: Changes are done.

File: web_security_dialog/static/src/js/web_security_dialog.js (hunk 6 above)
Comment (vishal-onnet) [reply web_security_dialog/static/src/js/web_security_dialog.js]: Changes are done.

File: web_security_dialog/static/src/js/web_security_dialog.js (hunk 7 above)
Comment (vishal-onnet) [reply web_security_dialog/static/src/js/web_security_dialog.js]: Changes are done.

File: web_security_dialog/__manifest__.py (hunk 5 above)
Comment (vishal-onnet) [reply web_security_dialog/__manifest__.py]: Yes sir i will change it soon.

File: web_security_dialog/models/res_users.py (hunk 3 above)
Comment (vishal-onnet) [reply web_security_dialog/models/res_users.py]: Yes sir i will change it soon.

File: web_security_dialog/models/res_users.py (hunk 2 above)
Comment (vishal-onnet) [reply web_security_dialog/models/res_users.py]: Changes are done.

**Pull request merged**